client/lib/wp/sync-handler/cache-invalidation.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import { getLocalForage } from 'lib/localforage';
+
+/**
+ * Module variables
+ */
+const localforage = getLocalForage();
+const debug = debugFactory( 'calypso:sync-handler:cache' );
+const RECORDS_LIST_KEY = 'records-list';
+// set record lifetime to 2 days
+const LIFETIME = 1000 * 60 * 60 * 24 * 2;
+
+export class CacheInvalidation {
+	getAll( fn = () => {} ) {
```

Comment:
also why are we assigning a function to an undeclared variable inside another function call?